### ISSUE TITLE
handle PKCS#7 CompressedData

### ISSUE THREAD

**Issue by dkg** [jstedfast/gmime#66]
gmime currently has some commented out/unimplemented functions for handling PKCS#7 CompressedData messages.  These should be implemented, so i'm just recording that as an open issue.

The functions in particular are:

- `g_mime_application_pkcs7_mime_compress`
- `g_mime_application_pkcs7_mime_decompress`

for specification reference, see [§ 3.6 of RFC 8551](https://tools.ietf.org/html/rfc8551#section-3.6)

**Comment by jstedfast** [comment]
Yea, I think I aborted my efforts on support for this because I wasn't sure how to do it using GpgME.

**Comment by dkg** [comment]
I don't think it should require gpgme at all -- compression and decompression are completely unrelated to cryptography, they just happen to use the same PKCS#7 framing.

**Comment by jstedfast** [comment]
Sure, but I was hoping gpgme supported it because otherwise I need to implement asn1 myself.

*Maybe* I can use libksba, but that looks like it's missing some API that might be needed. Hard to tell right now.

I'll also need to add new virtual methods for this (theoretically), so this may be best as a v4.0 API as well.

**Comment by sakwaa** [comment]
Are there any updates to the implementation for the compress/decompress? I ran into the same issue.. I have a multipart/signed part with a compressed message and I can't figure out how to decompress to get the message contents as readable text. Thanks!

**Comment by jstedfast** [comment]
No updates. I haven't had enough free time to look into this yet.
